armi/reactor/blueprints/__init__.py

```diff
@@ -393,69 +394,77 @@ def _resolveNuclides(self, cs):
-
+        originalFlagKeys = set(self.nuclideFlags)
         isotopicOptions.autoUpdateNuclideFlags(cs, self.nuclideFlags)
 
+        nucsToForceInXsGen = set()
+        # just expanding flags now. ndense gets expanded in comp blueprints
         self.elementsToExpand = []
         for nucFlag in self.nuclideFlags:
             # this returns any nuclides that are flagged specifically for expansion by input
-            expandedElements = nucFlag.fileAsActiveOrInert(
-                actives, inerts, undefBurnChainActiveNuclides
+            (
+                expandedElements,
+                undefBurnChainActiveNuclides,
+            ) = nucFlag.fileAsActiveOrInert(
+                actives,
+                inerts,
             )
             self.elementsToExpand.extend(expandedElements)
 
         inerts -= actives
         self.customIsotopics = self.customIsotopics or isotopicOptions.CustomIsotopics()
-        (
-            elementalsToKeep,
-            expansions,
-        ) = isotopicOptions.autoSelectElementsToKeepFromSettings(cs)
+        eleKeep, eleExpand = isotopicOptions.eleExpandInfoBasedOnCodeENDF(cs)
 
         # Flag all elementals for expansion unless they've been flagged otherwise by
         # user input or automatic lattice/datalib rules.
-        for elemental in nuclideBases.instances:
-            if not isinstance(elemental, nuclideBases.NaturalNuclideBase):
+        for nucBase in nuclideBases.instances:
+            wasInOriginalFlags = nucBase.name in originalFlagKeys
+            isAlreadyIsotopic = not isinstance(nucBase, nuclideBases.NaturalNuclideBase)
+            if isAlreadyIsotopic:
+                if wasInOriginalFlags:
+                    nucsToForceInXsGen.add(nucBase.name)
                 # `elemental` may be a NaturalNuclideBase or a NuclideBase
-                # skip all NuclideBases
+                # skip all NuclideBases (isotopics)
                 continue
 
-            if elemental in elementalsToKeep:
+            # we now know its an elemental
+            elemental = nucBase
+            if elemental in eleKeep:
+                if wasInOriginalFlags:
+                    nucsToForceInXsGen.add(elemental.name)
                 continue
-
             if elemental.name in actives:
                 currentSet = actives
-                actives.remove(elemental.name)
             elif elemental.name in inerts:
                 currentSet = inerts
-                inerts.remove(elemental.name)
             else:
                 # This was not specified in the nuclide flags at all.
                 # If a material with this in its composition is brought in
                 # it's nice from a user perspective to allow it.
                 # But current behavior is that all nuclides in problem
                 # must be declared up front.
+                if wasInOriginalFlags:
+                    nucsToForceInXsGen.add(elemental.name)
```

Comment:
I don't know if I follow how we get to line 449. If `wasInOriginalFlags == true`, it should be in `actives` or `inerts`, yes?

The line doesn't do any damage but I'm just curious whether it's reachable. Maybe there's a situation I haven't thought of.

Reply:
yeah not sure its reachable either. i can just delete. its not in xs if so so we shouldn't force.